.github/workflows/tests-ces-migrate.yaml

```diff
@@ -0,0 +1,240 @@
+name: CiliumEndpointSlice migration (ci-ces-migrate)
+
+# Any change in triggers needs to be reflected in the concurrency group.
+on:
+  pull_request: {}
```

Comment:
Interesting rebuttal :grin: All good points and agreed, I think most of my opinions are based on my own assumptions. Sounds like we might not want to use Ariane here.

> I've typically thought of it as a way to protect against abuse

My POV on this was more about wasting resources than abusing, but this could be my own bias. I use CI to smoke test draft PRs fairly often, it's usually inevitable that I forget to run one a make target and there's a lingering typo or a change to a helm value. I could be an outlier on this one, though. I assumed we were trying to move E2E/long-running tests mostly to Ariane.

> The workflow seems to be running fine directly from the PR

Yes, I think I may have been conflating this with workflows we had that ran on `pull_request_target`

Reply:
I agree with the sentiment that we may not need to run this test on every PR - most code changes would not impact this test. Currently we largely only have tests that run when PRs are posted and tests that are run upon merge. However if something fails on merge then the triage process is largely ad-hoc for correlating the failure back to the PR that triggers the failure.

The test run currently on this PR seems to take <20m, while for smoke tests we'd ideally be <10m (on this PR: 6-7m). So maybe it's not a bad idea to switch this over to being part of the `/test` jobs after all.

Reply:
^ I think the main downside is that if we need to further change this workflow, it's simply going to be easier if it's based on `pull_request`. I'd like to get feedback from someone in @cilium/ci-structure on this, maybe @viktor-kurchenko or @nbusseneau could chime in too if they have thoughts on this.

Reply:
`pull_request` is definitely fine here, just like for our smoke tests this does not require access to any external resources. However if it does take a long time and also does not need to run on every PR, it's indeed a resource hog.

Personally I would say if we can be confident that this workflow only needs to run when changes are made to specific files / directories, then we can keep it `pull_request`-based and do the filtering in workflow, but if we cannot and considering we don't want `pull_request`-based workflows to take long, then we should move to Ariane even if annoying.

Reply:
Thanks all, this was all very helpful and interesting to learn! :) I'm wondering if one of you knows if it is possible to filter this workflow / the scenarios in which you would want to run this test? 

And if not, will it be possible for me to change to an Ariane /test workflow from this PR (from an external fork) or should it be approached differently? I may still be a bit confused about the cause of the problem with testing using that method from earlier.

Reply:
Since this workflow is specific to the migration from CEP to CES, it seems to me like `pull_request` + filtering would be ideal. I think looking for changes in `pkg/k8s/apis/cilium.io` and `operator/pkg/ciliumendpointslice` would give us enough coverage for the migration.

Reply:
Got it, thank you!